### PR TITLE
Implement FetchX509Bundles and support unregistered retrievals of x509 bundles 

### LINKED ISSUE #2089: Implement FetchX509Bundles support
The Workload API recently gained an RPC to return the set of X.509 bundles. This is intended for workloads who are only in the business of validating X.509 SVIDs, but don't need one themselves.

This issue tracks the implementation of FetchX509Bundles in SPIRE.

Here are the rough steps to make this happen:
1) Update the Workload API definition in go-spiffe v2 (SPIRE gets the definition and codegen from here)
2) Implement the RPC handler
3) Allow authorization to be optional (default to required, override via a flag). If a workload is not authorized (i.e. it does not match with one or more registration entries), then only return the bundle for the trust domain of the agent, i.e. do NOT include federated bundles. This follows the thinking in #1834.

### PR DESCRIPTION
**Pull request by srwaggon**
**Pull Request check list**

- [x] Commit conforms to CONTRIBUTING.md?
- [x] Proper tests/regressions included?
- [x] Documentation updated?
- [x] Include DCO

**Affected functionality**

Implement the FetchX509Bundles rpc to allow workload api clients to easily get updates for the x509 bundles.

**Description of change**

We introduced a new configuration flag. This flag is:
`-allowUnregisteredClientsToFetchX509Bundles` as a command flag or
`allow_unregistered_clients_to_fetch_x509_bundles` in the configuration file.

See test/fixture/config/agent_good.conf for an example conf file usage.

This configuration will allow clients that do not have a workload registration entry to retrieve x509 bundles from the spire-agent using the WorkloadAPI.

These unregistered clients will only receive the bundles for the trust domain of the agent.

**Which issue this PR fixes**

fixes #2089 

There are still a few things we need to do for this changeset. We are hoping to get some feedback on the naming while we polish it up. 

The config/flag names are `allow_unregistered_clients_to_fetch_x509_bundles`/`allowUnregisteredClientsToFetchX509Bundles`. Do these names sound like the user experience that SPIRE is looking for? Do you have a preferred name for this property?

There is a temporary commit pointing to a forked go-spiffe in the go.mod. We are unable to use the upstream until the [go-spiffe RPC PR](https://github.com/spiffe/go-spiffe/pull/159) is merged.

- with @adobley

### CODE REVIEW

**Comment by amartinezfayo** [comment]
Thank you @srwaggon for this contribution!

> There are still a few things we need to do for this changeset. We are hoping to get some feedback on the naming while we polish it up.
> 
> The config/flag names are allow_unregistered_clients_to_fetch_x509_bundles/allowUnregisteredClientsToFetchX509Bundles. Do these names sound like the user experience that SPIRE is looking for? Do you have a preferred name for this property?

I'm not sure if "unregistered clients" is the best way to refer to unauthenticated workloads. The term "client" may be confusing for example if it's a server that is making the request. Also, the workload may be "registered" but still not be able to retrieve an identity based on the selectors resolved. So to me, referring to "authenticated / unauthenticated" would be more precise.
I think that something along the lines of `allow_unauthenticated_verifiers` as suggested by @azdagron in #1834 is a better choice for this, that can be used for FetchX509Bundles and (eventually) FetchJWTBundles.

**Comment by srwaggon** [comment]
(with -@adobley)

We've updated the flag name from `-allowUnregisteredClientsToFetchX509Bundles` to `-allowUnauthenticatedVerifiers` as per suggestion. 

We have also updated the documentation to reflect this new flag.
We are also still awaiting the merging of the [RPC for fetching x509 bundles](https://github.com/spiffe/go-spiffe/pull/159) so that the go.mod can be correctly updated.

**Comment by adobley** [comment]
https://github.com/spiffe/go-spiffe/pull/159 was merged so we've updated the commits to reference the new go-spiffe v2.0.0-beta.5 tag.

This changeset should be good to go!

@srwaggon can you change this from a draft when you have the time? Thanks!

File: pkg/agent/endpoints/workload/handler.go [hunk 1]
```diff
@@ -205,6 +206,65 @@ func (h *Handler) FetchX509SVID(_ *workload.X509SVIDRequest, stream workload.Spi
...
+		log.WithError(err).Error("Failed to send X509 bundle response")
+		return err
+	}
+
+	return nil
+}
+
+func composeX509BundlesResponse(update *cache.WorkloadUpdate) *workload.X509BundlesResponse {
+	bundles := make(map[string][]byte)
+	if update.Bundle != nil {
```
Comment (MarcosDY) [inline pkg/agent/endpoints/workload/handler.go:253]: based on SPEC ```The `X509BundlesResponse` response message has a mandatory `bundles` field, which MUST contain at least the trust bundle for the trust domain in which the server resides. The `crl` field is optional.```
if Bundle is nil, may we return an error?

**Review by azdagron — APPROVED** [review]
Thank you for this @adobley and @srwaggon! I've put this through it's paces and it looks good.

**Comment by azdagron** [comment]
Looks like the update broke the PR. I'll push a commit to patch.

**Comment by azdagron** [comment]
Also added the defensive check while I was in there.

**Review by MarcosDY — APPROVED** [review]
LGTM!

**Review by azdagron — APPROVED** [review]
Thanks for taking a look, @MarcosDY !

**Pull request merged**